### DISCUSSION TITLE
Missing release notes for latest release v3.0.0-beta-2036

### DISCUSSION THREAD

**Discussion by viceice** [sebastienros/jint#1003] [General]
Hi, the release notes for `v3.0.0-beta-2036` are missing.
I've seen some compile time breakings. 🙃 

https://github.com/sebastienros/jint/releases/tag/v3.0.0-beta-2036

**Comment by lahma** [comment]
I've created the missing release notes: https://github.com/sebastienros/jint/releases/tag/v3.0.0-beta-2036 , thanks for reporting!

What broke for you?

**Reply by viceice** [reply]
`JSValue.JsValue` now needs `Engine` as first argument 😉 Was easy to fix, as it was deprecated.

**Reply by viceice** [reply]
and #994, but thats ok, as i don't want and don't need it. Just had a test for it, which now tests for throwing an error 🙃

**Reply by lahma** [reply]
Good to hear that nothing too bad, APIs can be hard to get right and I don't promise they are right even yet 😄

**Reply by viceice** [reply]
Sure, no worry, I know I'm using a `beta` version. So I expect API changes. My CI will catch the build issues, so i can fix them.

Good work so far, like to see working `import / export` esm working, so i can switch back from Nil.JS to JInt 🚀

**Reply by lahma** [reply]
It's a [work in progress](https://github.com/sebastienros/jint/pull/990) 🙂  may I ask what makes you switch between the two, pros/cons? I know Nil.JS is also a fine library.

**Reply by viceice** [reply]
*Nil.JS*
Has full es6 support, but is lacking newer es7 and es8 syntactic sugars. Also there is only one maintainer, so small bugs need sometimes 2 or morgen weeks to get fixed, even a community PR is open.

Also the Promises are implemented via tasks and it looks like it allows parallel tasks, which isn't expected from a nodejs perspective. They should run in a random sequence.

*Jint*
Is missing the module support, my major blocker, as we use modern es6 code for server side scripting.

Also jint has some modern es Operators which makes code a lot more readable. 😅

**Reply by sebastienros** [reply]
> They should run in a random sequence

I believe it's less random in Jint, it will try to dequeue previous stacks from Promises first if they are available. AFAIR. 

> Is missing the module support

Very eager to test this promising PR too.

**Reply by lahma** [reply]
Thanks for the honest feedback, I hope Jint can soon deliver things you expect!

**Reply by viceice** [reply]
> > They should run in a random sequence
> 
> I believe it's less random in Jint, it will try to dequeue previous stacks from Promises first if they are available. AFAIR. 

This is no problem, only requirement is to not run in parallel on different threads, as a lot objects are not thread safe in dotnet. 😉


> 
> > Is missing the module support
> 
> Very eager to test this promising PR too. 

I'll wait for merge. 🤭
